File: 305 - C++/HW5/Shapes/2DShapes/Rectangle.cpp
#include <iostream>
#include "Rectangle.h"

double Rectangle::computeArea() {
	return width * height;
}

void Rectangle::printShapeDetail() {
	std::cout << "Rectangle, 2D Shape, width: " << width
	          << ", height: " << height
	          << ", Area: " << computeArea()
	          << "." << std::endl;
}

File: 349 - Design Patterns/FactoryPattern/src/shape/Rectangle.java
package shape;

class Rectangle extends Shape implements Comparable {
	Rectangle(String name, double height, double width) {
		super(name, height, width);
	}

	Rectangle(double height, double width) {
		super("Rectangle", height, width);
	}

	@Override
	public double calculateArea() {
		return height * width;
	}
}
